h2/src/main/org/h2/command/dml/Select.java

```diff
@@ -1086,14 +1086,23 @@ public String getPlanSQL() {
         StatementBuilder buff = new StatementBuilder();
         for (TableFilter f : topFilters) {
             Table t = f.getTable();
-            if (t.isView() && ((TableView) t).isRecursive()) {
-                buff.append("WITH RECURSIVE ").append(t.getName()).append('(');
-                buff.resetCount();
-                for (Column c : t.getColumns()) {
-                    buff.appendExceptFirst(",");
-                    buff.append(c.getName());
+            TableView tableView = (t instanceof TableView) ? (TableView) t : null;
+            if (tableView!=null && tableView.isView() && tableView.isRecursive() && tableView.isTableExpression()) {
```

Comment:
tableView!=null && tableView.isView()  is redundant, one or the other is fine

Reply:
Done